src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -383,12 +394,47 @@ mir::shell::SurfaceSpecification& mf::WindowWlSurfaceRole::spec()
     return *pending_changes;
 }
 
+void mf::WindowWlSurfaceRole::apply_pending_size_bounds()
+{
+    if (weak_scene_surface.lock())
+    {
+        if (pending_changes)
+        {
+
+#define SET_IF_CHANGED(pending, committed)  \
+    if (pending)                            \
+    {                                       \
+        if (pending.value() == committed)   \
+            pending.consume();              \
+        else                                \
+            committed = pending.value();    \
```

Comment:
Should this also be `committed = pending.consume()`?

Reply:
C++ allows this to be written as an inline function template. No need for a macro. (I'm not sure that it can be done as a lambda for 16.04, but internal linkage at namespace scope would work for me.)

@RAOF I think you misread the intent: we don't reset on a change as we later rely on `pending_changes->is_empty()` to see if we applied changes. (The names don't help with following the code.)